src/app/loadout-builder/generated-sets/SetStats.tsx

```diff
@@ -91,7 +93,7 @@ function SetStats({
           )}
         >
           <Stat
-            isActive={enabledStats.has(statHash)}
+            isActive={statConstraints.some((s) => s.statHash === statHash)}
```

Comment:
This is less efficient, but I'm not sure it's worth it to precalculate an `enabledStats` set just for this sort of thing.